runtime-components/non-repudiation-postgresql/src/main/scala/com/daml/nonrepudiation/postgresql/PostgresqlCertificateRepository.scala

```diff
@@ -25,7 +25,7 @@ final class PostgresqlCertificateRepository(transactor: Transactor[IO])(implicit
     getCertificate(fingerprint).query[X509Certificate].option.transact(transactor).unsafeRunSync()
 
   private def putKey(fingerprint: FingerprintBytes, certificate: X509Certificate): Fragment =
-    fr"insert into" ++ table ++ fr"""(fingerprint, certificate) values ($fingerprint, $certificate)"""
+    fr"insert into" ++ table ++ fr"""(fingerprint, certificate) values ($fingerprint, $certificate) on conflict do nothing"""
```

Comment:
Answering my own question: We don’t have to worry about someone trying to insert a certificate with the same fingerprint but a different certificate because we compute the fingerprint ourselve and don’t let users specify it.